dlt/destinations/impl/filesystem/factory.py

```diff
@@ -46,12 +52,27 @@ def __init__(
             credentials: Credentials to connect to the filesystem. The type of credentials should correspond to
                 the bucket protocol. For example, for AWS S3, the credentials should be an instance of `AwsCredentials`.
                 A dictionary with the credentials parameters can also be provided.
+            current_datetime: current datetime used instead of datetime generated by dlt.
+            datetime_format: strftime formatting for current_datetime
+            extra_params: custom layout parameters, all unknown parameters will be skipped,
+                values can be primitive types or callables which also should return a primitive type.
             **kwargs: Additional arguments passed to the destination config
         """
+        destinatination_kwargs: t.Dict[str, t.Any] = {}
+        if current_datetime:
+            destinatination_kwargs["current_datetime"] = current_datetime
+
+        if datetime_format:
+            destinatination_kwargs["datetime_format"] = datetime_format
+
+        if extra_params:
+            destinatination_kwargs["extra_params"] = extra_params
```

Comment:
I am passing these three arguments via `kwargs` so [here](https://github.com/dlt-hub/dlt/pull/1149/files#diff-650cdec3d48ccec34c894eb426306aeabc9dccae1d3872850666abe396d53024R64) on resolve I reconcile and can override the values from `config.toml`.
It doesn't feel right however is it fine @sh-rp @rudolfix ?

Reply:
Done.